python/cugraph/cugraph/tests/generators/test_rmat.py

```diff
@@ -40,27 +80,269 @@ def _call_rmat(scale, num_edges, create_using, mg=False):
+                include_edge_types=include_edge_types,
+                min_edge_type=min_edge_type,
+                max_edge_type=max_edge_type,
+            )
+
+            # Check that there is an 'edge_type' column
+            assert "edge_type" in df.columns
+            edge_types_err1 = df.query("{} < edge_type".format(max_edge_type))
+            edge_types_err2 = df.query("{} > edge_type".format(min_edge_type))
+
+            # Check that edge weights values are between 'min_edge_type'
+            # and 'max_edge_type'.
+            assert len(edge_types_err1) == 0
+            assert len(edge_types_err2) == 0
+    else:
+        df = _call_rmat(
+            scale,
+            num_edges,
+            create_using,
+            scramble_vertex_ids=scramble_vertex_ids,
+            include_edge_types=include_edge_types,
+            min_edge_type=min_edge_type,
+            max_edge_type=max_edge_type,
+        )
+        assert len(df.columns) == 2
+
+
+@pytest.mark.sg
+@pytest.mark.filterwarnings("ignore:make_current is deprecated:DeprecationWarning")
+@pytest.mark.parametrize("scale", [2, 4, 8], ids=_scale_test_ids)
+@pytest.mark.parametrize(
+    "include_edge_weights", _include_edge_weights, ids=_include_edge_weights_test_ids
+)
+@pytest.mark.parametrize("clip_and_flip", _clip_and_flip, ids=_clip_and_flip_test_ids)
+@pytest.mark.parametrize(
+    "scramble_vertex_ids", [False], ids=[_scramble_vertex_ids_test_ids[0]]
+)
+def test_rmat_clip_and_flip(
+    scale, include_edge_weights, clip_and_flip, scramble_vertex_ids
+):
     """
+    Verifies that there are edges only in the lower triangular part of
+    the adjacency matrix when 'clip_and_flip' is set to 'true'.
+
+    FIXME: 'scramble_vertex_ids' nullifies the effect of 'clip_and_flip' therefore
+    both flags should not be set to 'True'.
 
+    """
     num_edges = (2**scale) * 4
     create_using = None  # Returns the edgelist from RMAT
+    minimum_weight = 0
+    maximum_weight = 1
+    dtype = "FLOAT32"
+    df = _call_rmat(
+        scale,
+        num_edges,
+        create_using,
+        clip_and_flip=clip_and_flip,
+        scramble_vertex_ids=scramble_vertex_ids,
+        include_edge_weights=include_edge_weights,
+        dtype=dtype,
+        minimum_weight=minimum_weight,
+        maximum_weight=maximum_weight,
+    )
+
+    if not include_edge_weights:
+        df["weights"] = 1
+        # cupy coo_matrix only support 'float32', 'float64', 'complex64'
+        # and 'complex128'.
+        df["weights"] = df["weights"].astype("float32")
 
-    df = _call_rmat(scale, num_edges, create_using)
-    assert len(df) == num_edges
+    dim = df[["src", "dst"]].max().max() + 1
+    src = df["src"].to_cupy()
+    dst = df["dst"].to_cupy()
+    weights = df["weights"].to_cupy()
+    adj_matrix = coo_matrix((weights, (src, dst)), shape=(dim, dim)).toarray()
+
+    upper_coo = triu(adj_matrix)
+    diag = tril(upper_coo)
+
+    if clip_and_flip:
```

Comment:
We need to tests `clip_and_flop` to make sure the edges are indeed in the lower triangle part of the adjacency matrix when set to `True` and this is what this tests is doing. In case it is `False`, the test makes sure the run doesn't crash (we can't do much since there is nothing else we can verify)